### PR TITLE
[Java] Fix some potential bugs about `Ray.shutdown()`

### PR DESCRIPTION
**Pull request by kfstorm**
## Why are these changes needed?

In Java tests, if `Ray.init()` failed in `BaseTest.setUpBase`, the `BaseTest.tearDownBase` will also fail due to `filesToDelete` is null.

This PR also adds statements to set some variables to null during the shutdown process. And it makes sure the `ExecutorService` in `LocalModeTaskSubmitter` will be shut down.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.

### CODE REVIEW

File: java/test/src/main/java/org/ray/api/test/BaseTest.java [hunk 1]
```diff
@@ -37,9 +37,10 @@ public void setUpBase(Method method) {
   @AfterMethod
   public void tearDownBase() {
     Ray.shutdown();
-
-    for (File file : filesToDelete) {
-      file.delete();
+    if (filesToDelete != null) {
```
Comment (raulchen) [inline java/test/src/main/java/org/ray/api/test/BaseTest.java]: better just initialize `filesToDelete` to an empty list.

Comment (raulchen) [reply java/test/src/main/java/org/ray/api/test/BaseTest.java]: `BaseMultiLanguageTest` also has this issue, can you also fix that?

**Review by raulchen — CHANGES_REQUESTED** [review]

File: java/test/src/main/java/org/ray/api/test/BaseTest.java (hunk 1 above)
Comment (kfstorm) [reply java/test/src/main/java/org/ray/api/test/BaseTest.java]: Updated.

PS: `BaseMultiLanguageTest` doesn't have this issue.

Comment (kfstorm) [reply java/test/src/main/java/org/ray/api/test/BaseTest.java]: Sorry that I didn't notice that `BaseMultiLanguageTest` has another issue. Fixed.

**Review by raulchen — APPROVED** [review]

**Pull request merged**